### PR TITLE
GBMM: all precisions and tester. 

### PR DESCRIPTION
**Pull request by DSMishler**
Header files and makefile paths changed accordingly. 

Cleaned commit history, now properly on a branch, following PR etiquette. Previous comments that have been mentioned in review have been addressed!

Work on GBMM stretching back to ~July 2021. (now with cleaner modified history) - Major changes:

-     all gbmm functionality added (nonsquare, arbitrary kl,ku, all precisions, all transpose combinations)
-     gbmm tester added: also to all precisions. Format mimics that of gemm tester
-     relevant header files and python generating scripts modified to accommodate gbmm.

 

 Review would be most useful on:

 -     Double-checking that all supported build paths still work with gbmm (mainly CMake)
 -     Coding style. Force of habit may have allowed some coding style difference to remain.

Old untampered commit history of my work is stored on a "dummy" repo on my machine in case it is ever needed.

### CODE REVIEW

File: compute/pzgb2desc.c [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                                    // plasma_tile_mmain_band need not know m,n
+            x1 = n == 0 ? A.j%A.nb : 0;
+            y1 = m == 0 ? A.i%A.mb : 0;
+            x2 = n == A.nt-1 ? (A.j+A.n-1)%A.nb+1 : A.nb;
+            y2 = m == A.mt-1 ? (A.i+A.m-1)%A.mb+1 : A.mb;
+
+
+            f77 = &pA[(size_t)A.nb*lda*n + (size_t)A.mb*m];
+            bdl = (plasma_complex64_t*)plasma_tile_addr(A, m, n);
+            // printf("[%s] Full view:\nm=%d\nn=%d\nx1=%d\ny1=%d\nx2=%d\ny2=%d\nlda=%d\n",
```
Comment (luszczek) [inline compute/pzgb2desc.c:58]: I think this is a leftover debug code and should be removed.

File: compute/zgbmm.c [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+    // Initialize sequence.
+    plasma_sequence_t sequence;
+    retval = plasma_sequence_init(&sequence);
+
+    // Initialize request.
+    plasma_request_t request;
+    retval = plasma_request_init(&request);
+
+    // asynchronous block
+    // if considering debugging , remove these directives.
```
Comment (luszczek) [inline compute/zgbmm.c:249]: Please remove this comment. No information about debugging please.

**Review by luszczek — CHANGES_REQUESTED** [review]

File: compute/zgbmm.c (hunk 2 above)
Comment (DSMishler) [reply compute/zgbmm.c]: Done!

File: compute/pzgb2desc.c (hunk 1 above)
Comment (DSMishler) [reply compute/pzgb2desc.c]: Yes, you were correct. And there was another comment above it referring to debug information - it was also removed

**Review by luszczek — CHANGES_REQUESTED** [review]

File: compute/pzgb2desc.c [hunk 3]
```diff
@@ -36,13 +36,11 @@ void plasma_pzgb2desc(plasma_complex64_t *pA, int lda,
     int n, m, ldt;
     for (m = 0; m < A.mt; m++) {
         for (n = 0; n < A.nt; n++) {
-            // don't want to copy tiles without elements (plasma_tile_addr)
-            // cannot handle it.
+            // don't want to copy tiles without elements because
+            // (plasma_tile_addr) cannot handle it.
             // Calculate kut without normal space for transformations.
```
[the remark this answers is not in the document]
Comment (DSMishler) [reply compute/pzgb2desc.c:42]: `kut` did belong in that comment - it just wasn't clear that what I was doing was replacing what would normally be referred to as `kut`. Adding a commit with this change.

**Review by luszczek — APPROVED** [review]
Thank you for your contribution.

**Pull request merged**